sklearn/tree/meson.build

```diff
@@ -16,7 +16,7 @@ tree_extension_metadata = {
 foreach ext_name, ext_dict : tree_extension_metadata
   py.extension_module(
     ext_name,
-    ext_dict.get('sources'),
+    ext_dict.get('sources') + utils_cython_tree,
```

Comment:
From looking at the other changes, `py.extension_module` accepts a list of list. If that's the case, can we keep how we call `py.extension_module` consistent:

```suggestion
    [ext_dict.get('sources'), utils_cython_tree],
```